utxo/src/cache.rs

```diff
@@ -351,16 +200,12 @@ impl<'a> UtxosCache<'a> {
         // self.memory_usage must be deducted from this entry's size
 
         // check whether this entry is fresh
-        if entry.is_fresh {
+        if entry.is_fresh() {
             // This is only available in this view. Remove immediately.
             self.utxos.remove(outpoint);
         } else {
             // mark this as 'spent'
-            let new_entry = UtxoEntry {
-                status: UtxoStatus::Spent,
-                is_dirty: true,
-                is_fresh: false,
-            };
+            let new_entry = UtxoEntry::new(None, IsFresh::No, IsDirty::Yes);
```

Comment:
One thing just occurred to me. Are all the combinations possible here? E.g. is it possible to have UTXO freash and dirty simultaneously? If not, maybe one enum (like `UtxoState` or something) would be better so only valid states can be constructed.

Reply:
Good observation! Actually out of these 2 fields every combination is possible, but if we take 3rd field `status` into account then only 5 out of 8 is valid. I'll add assertions into constructor

Reply:
Would it be possible to reorganize the enums so the invalid states cannot be constructed? That would move the run-time checking (assertions) to compilation time, which is preferable unless it makes the code too cryptic.

Reply:
yeah the problem is that code starts looking weird. As a compromise `debug_assert` can be used. `UtxoEntry` is private so no one outside the crate will construct it and all combinations can be checked by tests.

Reply:
Ok. Let's keep `assert` then